src/bin/nydusd/fs_cache.rs

```diff
@@ -642,19 +683,28 @@ impl FsCacheHandler {
                 );
                 return;
             }
-            Some((FsCacheObject::DataBlob(blob), u)) => {
+            Some((FsCacheObject::DataBlob(fsblob), u)) => {
                 fd = u;
-                match blob.get_blob_object() {
-                    None => {
-                        warn!("fscache: internal error: cached object is not BlobCache objects");
+                let guard = fsblob.read().unwrap();
+                match guard.get_blobcache() {
+                    Some(blob) => {
+                        match blob.get_blob_object() {
+                            None => {
+                                warn!("fscache: internal error: cached object is not BlobCache objects");
+                            }
+                            Some(obj) => match obj.fetch_range_uncompressed(msg.off, msg.len) {
+                                Ok(_) => {}
+                                Err(e) => error!(
+                                    "{}",
+                                    format!("fscache: failed to read data from blob object: {}", e,)
+                                ),
+                            },
+                        }
+                    }
+                    _ => {
+                        warn!("fscache: blob object not ready")
+                        //TODO: maybe we should retry init blob object here
```

Comment:
It would be better to add a flag to mark the init_blobcache() is still working or has failed out. If it's still working, we should wait for a while.

Reply:
Since Xin Yin is currently busy this week, could we merge this in advance and apply a follow-up fix later if possible?